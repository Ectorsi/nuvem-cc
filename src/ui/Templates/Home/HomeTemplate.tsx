import React from "react";
import PageTitle from "../../components/PageTitle/PageTitle";

import * as S from "./styles";
import PokeList, { PokeListProps } from "../../components/PokeList/PokeList";
import PokeFilter, { PokeFilterProps } from "../../components/PokeFilter/PokeFilter";
import PokemonDetailModal from "../../Modals/PokemonDetailModal/PokemonDetailModal";
import { GetPokemonDetailReturn } from "../../../domains/pokemon/types";

export type HomeTemplateProps = {
    title: string
    pokeList: PokeListProps
    pokemonDetails: GetPokemonDetailReturn | null
    showModal: boolean
    handleShowModal: () => void
    pokeFilter: PokeFilterProps
}

const HomeTemplate = ({ title, pokeList, pokemonDetails, showModal, handleShowModal, pokeFilter }: HomeTemplateProps) => {
    return (
        <>
            <S.Container>
                <PageTitle title={title} />
                <PokeFilter {...pokeFilter} />

                {/* 
                esse pokelist tem um clique que eh dentro do pocke card. 
                Esse click vai passar o nome do pokemon via callback handler. 
                essa logica vai estar em um custom hook que vai um state que vai armazenar o nome do pokemon que foi clicado.
            */}
                <PokeList
                    pokemonList={pokeList.pokemonList}
                    pokeCard={pokeList.pokeCard}
                    loadingPokemonsList={pokeList.loadingPokemonsList}
                    errorFetchPokemons={pokeList.errorFetchPokemons}
                />
            </S.Container>
            <PokemonDetailModal
                // {...pokemonDetails}
                handleShowModal={handleShowModal}
                showModal={showModal}
                pokemonDetails={pokemonDetails}

            />
        </>
    );
};

export default HomeTemplate;