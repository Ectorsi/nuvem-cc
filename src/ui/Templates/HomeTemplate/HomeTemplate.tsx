import PageTitle from '../../components/PageTitle/PageTitle';

import * as S from './styles';
import PokeList, { PokeListProps } from '../../components/PokeList/PokeList';
import PokeFilter, {
    PokeFilterProps,
} from '../../components/PokeFilter/PokeFilter';
import PokemonDetailModal from '../../Modals/PokemonDetailModal/PokemonDetailModal';
import { GetPokemonDetailReturn } from '../../../domains/pokemon/types';

export type HomeTemplateProps = {
    title: string;
    pokeList: PokeListProps;
    pokemonDetails: GetPokemonDetailReturn | null;
    showModal: boolean;
    handleShowModal: () => void;
    pokeFilter: PokeFilterProps;
    isPokemonDetailsLoading: boolean;
};

const HomeTemplate = ({
    title,
    pokeList,
    pokemonDetails,
    showModal,
    handleShowModal,
    pokeFilter,
    isPokemonDetailsLoading,
}: HomeTemplateProps) => {
    return (
        <>
            <S.Container>
                <PageTitle title={title} />
                <PokeFilter {...pokeFilter} />
                <PokeList {...pokeList} />
            </S.Container>
            <PokemonDetailModal
                handleShowModal={handleShowModal}
                showModal={showModal}
                pokemonDetails={pokemonDetails}
                isPokemonDetailsLoading={isPokemonDetailsLoading}
            />
        </>
    );
};

export default HomeTemplate;
