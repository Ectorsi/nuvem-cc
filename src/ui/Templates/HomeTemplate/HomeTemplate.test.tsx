import React from "react";
import { render, screen } from "@testing-library/react";

import HomeTemplate, { HomeTemplateProps } from './HomeTemplate';

const MockHomeTemplate: HomeTemplateProps = {
    title: 'Pokedex',
    pokeList: {
        pokemonList: [
            {
                name: 'Pikachu',
                url: 'www.test.com.br',
            },
            {
                name: 'Pikachu2',
                url: 'www.test.com.br',
            }
        ],
        pokeCard: {
            handleSelectPokemon: jest.fn()
        },
        errorFetchPokemons: '',
        loadingPokemonsList: false,
    },
    pokeFilter: {
        cleanFilters: jest.fn(),
        inputTextProps: {
            placeholder: 'Pesquisar',
            onChange: jest.fn(),
            value: 'Pikachu',
            error: '',
        },
        selectBoxRegion: {
            label: 'Tipo',
            placeholder: 'Selecione um tipo',
            value: 'Teste',
            onChange: jest.fn(),
            options: [
                {
                    value: 'Value Test',
                    label: 'Label test',
                },
                {
                    value: 'Value Test2',
                    label: 'Label test2',
                }
            ],
        },
        selectBoxLocation: {
            label: 'Geração',
            placeholder: 'Selecione uma geração',
            value: '',
            onChange: jest.fn(),
            options: [],
            error: '',
        },
        selectBoxAreas: {
            label: 'Espécie',
            placeholder: 'Selecione uma espécie',
            value: '',
            onChange: jest.fn(),
            options: [],
            error: '',
        },
    },
    pokemonDetails: {
        name: 'pikachu',
        abilities: [
            {
                ability: {
                    name: "static",
                    url: "https://pokeapi.co/api/v2/ability/9/"
                },
                is_hidden: false,
                slot: 1
            },
            {
                ability: {
                    name: "lightning-rod",
                    url: "https://pokeapi.co/api/v2/ability/31/"
                },
                is_hidden: true,
                slot: 3
            },
        ],
        height: 4,
        sprites: {
            front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
        },
        weight: 60
    },
    showModal: false,
    handleShowModal: jest.fn(),
};

describe('HomeTemplate integration test', () => {
    it('should be able to show all components', () => {
        const { container } = render(<HomeTemplate {...MockHomeTemplate} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should be able to show the title', () => {
        render(<HomeTemplate {...MockHomeTemplate} />);
        const title = screen.getAllByText(/Pokedex/i)[0];
        expect(title).toBeInTheDocument();
    });

    it('should be able to show the PokeList', () => {
        render(<HomeTemplate {...MockHomeTemplate} />);
        const pokeList = screen.getAllByText(/Pikachu/i)[0];
        expect(pokeList).toBeInTheDocument();
    });

    it('should be able to show the PokeFilter', () => {
        render(<HomeTemplate {...MockHomeTemplate} />);
        const pokeFilter = screen.getByPlaceholderText(/Pesquisar/i);
        expect(pokeFilter).toBeInTheDocument();
    });

    it('should be able to show the PokemonDetails', () => {
        render(<HomeTemplate {...MockHomeTemplate} />);
        const pokemonDetails = screen.getAllByText(/pikachu/i)[0];
        expect(pokemonDetails).toBeInTheDocument();
    });

    it('should be able to show the Modal', () => {
        render(<HomeTemplate {...MockHomeTemplate} />);
        const modal = screen.getAllByText(/x/i)[0];
        expect(modal).toBeInTheDocument();
    });
});
