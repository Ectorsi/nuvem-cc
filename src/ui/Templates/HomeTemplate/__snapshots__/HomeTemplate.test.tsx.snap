// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeTemplate integration test should be able to show all components 1`] = `
<div
  class="sc-gueYoa gkUGJR"
>
  <h1
    class="sc-beqWaB ekCqDZ"
  >
    Pokedex
  </h1>
  <div
    class="sc-fsQiph eyQCUQ"
  >
    <h2>
      Pesquise o Pokemon pelo nome
    </h2>
    <input
      class="sc-jsMahE dPXcAO"
      placeholder="Pesquisar"
      type="text"
      value="Pikachu"
    />
    <h2>
      Filtrar pokemons por região/área
    </h2>
    <p>
      Para filtrar um pokemon por regiao/area, você deve preencher todos os filtros!
    </p>
    <div
      class="sc-qRumB eTaYHB"
    >
      <div
        class="sc-csCMJt kjisHW"
      >
        <label
          class="sc-brKeYL cVRNkM"
        >
          Tipo
        </label>
        <select
          class="sc-fbJfA hFeYFK"
        >
          <option
            value=""
          >
            Selecione um tipo
          </option>
          <option
            value="Value Test"
          >
            Label test
          </option>
          <option
            value="Value Test2"
          >
            Label test2
          </option>
        </select>
      </div>
      <div
        class="sc-csCMJt kjisHW"
      >
        <label
          class="sc-brKeYL cVRNkM"
        >
          Geração
        </label>
        <select
          class="sc-fbJfA hFeYFK"
        >
          <option
            value=""
          >
            Selecione uma geração
          </option>
        </select>
      </div>
      <div
        class="sc-csCMJt kjisHW"
      >
        <label
          class="sc-brKeYL cVRNkM"
        >
          Espécie
        </label>
        <select
          class="sc-fbJfA hFeYFK"
        >
          <option
            value=""
          >
            Selecione uma espécie
          </option>
        </select>
      </div>
      <button
        class="sc-hAtEyd bfJMZR"
      >
        Limpar filtros
      </button>
    </div>
  </div>
  <div
    class="sc-dmqHEX bpogQo"
    id="list-items"
  >
    <button
      class="sc-jTrPJq ggUHYc"
      type="button"
    >
      <h1>
        Pikachu
      </h1>
    </button>
    <button
      class="sc-jTrPJq ggUHYc"
      type="button"
    >
      <h1>
        Pikachu2
      </h1>
    </button>
  </div>
</div>
`;
